librz/asm/asm.c

```diff
@@ -725,12 +725,17 @@ RZ_API int rz_asm_assemble(RzAsm *a, RzAsmOp *op, const char *buf) {
 	if (a->cur) {
 		Ase ase = NULL;
 		if (!a->cur->assemble) {
-			/* find callback if no assembler support in current plugin */
-			ase = findAssembler(a, ".ks");
-			if (!ase) {
-				ase = findAssembler(a, ".nz");
+			// Check if the syntax is GAS/AT&T.
+			if (a->syntax == RZ_ASM_SYNTAX_ATT) {
+				ase = findAssembler(a, ".as");
+			} else {
+				/* find callback if no assembler support in current plugin */
+				ase = findAssembler(a, ".ks");
 				if (!ase) {
-					ase = findAssembler(a, NULL);
+					ase = findAssembler(a, ".nz");
+					if (!ase) {
```

Comment:
Bring this `if` condition one level outside, so as to avoid deep nesting. Something like

```c
if (!ase) {
	ase = ...;
}
if (!ase) {
	ase = ...;
}
```